packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -258,4 +261,23 @@ export type SummaryElementStringifier = (contents: unknown) => string;
 /**
  * Parses a serialized/summarized string into an object, rehydrating any Fluid handles as necessary
  */
- export type SummaryElementParser = (contents: string) => unknown;
+export type SummaryElementParser = (contents: string) => unknown;
+
+/** Compose an {@link IChannelStorageService} which prefixes all paths before forwarding them to the original service */
+function scopeStorageService(service: IChannelStorageService, ...pathElements: string[]): IChannelStorageService {
+    function scopePath(path: string): string {
+        return `${pathElements.join("/")}/${path}`;
+    }
```

Comment:
I can't imagine it mattering perf-wise, but since we can expect the `scopePath` function to be called at least once, it may make more sense to eagerly compute the `pathElements.join("/")` part and store it for later use instead of recomputing it.
Feel free to ignore.

Reply:
Yea you are right; it makes the code cleaner too (no nested function). We could also memoize the whole object that is returned from this function for each index, since it's being recreated once for every index on every call to `load`. But I'm not sure how to do that without keeping around another array or something and the code bloat doesn't seem worth it.